src/components/Image/index.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import {Image as RNImage} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import ONYXKEYS from '../../ONYXKEYS';
+import {defaultProps, propTypes} from './imagePropTypes';
+import RESIZE_MODES from './resizeModes';
+
+class Image extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            imageSource: undefined,
+        };
+    }
+
+    componentDidMount() {
+        this.configureImageSource();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.source === this.props.source) {
+            return;
+        }
+        this.configureImageSource();
+    }
+
+    configureImageSource() {
+        const source = this.props.source;
+        const isAuthTokenRequired = this.props.isAuthTokenRequired;
+        let imageSource = source;
+        if (typeof source !== 'number' && isAuthTokenRequired) {
+            const authToken = lodashGet(this.props, 'session.encryptedAuthToken', null);
+            imageSource = {uri: `${source.uri}?encryptedAuthToken=${encodeURIComponent(authToken)}`};
+        }
```

Comment:
Can we add a comment here saying right now there's a `react-native-web` bug prevent us from adding headers to images for web / desktop? And that's why we do this here, but we add `headers` in `index.native.js`?